admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/content/ContentAdminAction.java

```diff
@@ -1,126 +1,238 @@
+        List<String> list = mapping.get(key);
+        if (null == list) {
+            return null;
+        }
+        return String.join(",", list);
+    }
+
+    @Override
+    public String updateSystemParams() {
+        String result = this.updateSystemParams(true);
+        if (!result.equals(SUCCESS)) {
+            return result;
+        }
+        try {
+            Map<String, List<String>> mapping = new HashMap<>();
+            this.updateResourceMapping(mapping, this.getResourceAltMapping(), IResourceManager.ALT_METADATA_MAPPING_KEY);
+            this.updateResourceMapping(mapping, this.getResourceDescriptionMapping(), IResourceManager.DESCRIPTION_METADATA_MAPPING_KEY);
+            this.updateResourceMapping(mapping, this.getResourceLegendMapping(), IResourceManager.LEGEND_METADATA_MAPPING_KEY);
+            this.updateResourceMapping(mapping, this.getResourceTitleMapping(), IResourceManager.TITLE_METADATA_MAPPING_KEY);
+            this.getResourceManager().updateMetadataMapping(mapping);
+        } catch (Throwable t) {
+            logger.error("error in configSystemParams", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    private void updateResourceMapping(Map<String, List<String>> mapping, String csv, String key) {
+        List<String> mappingList = (!StringUtils.isBlank(csv)) ? Arrays.asList(csv.split(",")) : null;
+        if (null == mappingList) {
+            mapping.remove(key);
+        } else {
+            mapping.put(key, mappingList);
+        }
+    }
+
+    /**
+     * Effettua la richiesta di ricaricamento degli indici a servizio del motore
+     * di ricerca interno.
+     *
+     * @return Il codice del risultato dell'azione.
+     */
+    public String reloadContentsIndex() {
+        try {
+            String result = this.configSystemParams();
+            if (!result.equals(SUCCESS)) {
+                return result;
+            }
+            this.getSearchEngineManager().startReloadContentsReferences();
+            logger.info("Reload contents index started");
+        } catch (Throwable t) {
+            logger.error("error in reloadContentsIndex", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    /**
+     * Effettua la richeista di ricaricamento delle referenze dei contenuti. Le
+     * referenze sono rappresentate sia dalle repliche dei valori degli
+     * Attributi di Contenuto dichiarati ricercabili in apposita tabella
+     * (elementi a servizio degli erogatori di contenuti in lista) e che delle
+     * referenze tra Contenuti e Pagine, Risorse, Gruppi e Contenuti stessi in
+     * apposita tabella (elementi a servizio di controlli di autorizzazione e
+     * validazione).
+     *
+     * @return Il codice del risultato dell'azione.
+     */
+    public String reloadContentsReference() {
+        try {
+            String result = this.configSystemParams();
+            if (!result.equals(SUCCESS)) {
```

Comment:
Just to check is this return correct? If the first one fails you don't want the contentManager call?

Reply:
yes, it's only to check the first call; you could eliminate this check (very difficult for it to fail), I put it to be sure